cmd/ova-provider-server/ova-provider-server.go

```diff
@@ -378,14 +420,18 @@ func convertToVmStruct(envelope []Envelope, ovaPath []string) ([]VM, error) {
 					FileRef:                 disk.FileRef,
 					Format:                  disk.Format,
 					PopulatedSize:           disk.PopulatedSize,
+					Name:                    name,
 				})
+				newVM.Disks[j].ID = diskIDMap.GetUUID(newVM.Disks[j], ovaPath[i]+"/"+name)
+
 			}
 
-			for _, network := range vmXml.NetworkSection.Networks {
+			for j, network := range vmXml.NetworkSection.Networks {
 				newVM.Networks = append(newVM.Networks, VmNetwork{
 					Name:        network.Name,
 					Description: network.Description,
 				})
+				newVM.Networks[j].ID = networkIDMap.GetUUID(newVM.Networks[j].Name, network.Name)
```

Comment:
this is a bit confusing - we pass network.Name twice then, right? it would be more clear with something like:
```golang
                                newVM.Networks = append(newVM.Networks, VmNetwork{
					Name:        network.Name,
					Description: network.Description,
                                        ID:          networkIDMap.GetUUID(network.Name, network.Name)
				})
```

Reply:
yes for the network specific its redundant since we using only the name and not the entire struct.